libcontainer/rootfs_linux.go

```diff
@@ -1071,16 +1071,16 @@ func remount(m *configs.Mount, rootfs string, mountFd *int) error {
 		if err == nil {
 			return nil
 		}
-		// Check if the source has ro flag...
+		// Check if the source has ro, nodev, noexec, nosuid flag...
 		var s unix.Statfs_t
 		if err := unix.Statfs(source, &s); err != nil {
 			return &os.PathError{Op: "statfs", Path: source, Err: err}
 		}
-		if s.Flags&unix.MS_RDONLY != unix.MS_RDONLY {
+		if s.Flags&(unix.MS_RDONLY|unix.MS_NODEV|unix.MS_NOEXEC|unix.MS_NOSUID) == 0 {
```

Comment:
Hmm, why did you drop the `RDONLY` flag?

Reply:
What do you mean by drop? If `s.Flags` has none of the flags `unix.MS_RDONLY|unix.MS_NODEV|unix.MS_NOEXEC|unix.MS_NOSUID` set we error out. Previously we only did this if `unix.MS_RDONLY` was not set.

Reply:
Right, we are just adding only the flags that the mountpoint has (from those flags, not all the flags)